### ISSUE TITLE
Suggestion: Allow breakable frames

### ISSUE THREAD

**Issue by just-bqn** [marc-thieme/frame-it#1]
## Issue

Currently, when the text inside a frame is too long, it simply overflows the page.

```
#import "@preview/frame-it:1.0.0": *

#let (theorem,) = make-frames(
  "counter-id",
  theorem: ("Theorem",),
)

#theorem[A very long theorem][
  #lorem(1000)
]
```

![image](https://github.com/user-attachments/assets/17cf9469-19aa-4703-849b-fa03418dd0ce)

## Suggestion

Is it possible to implement a `breakable` option that when set to `true`, allows the extra text to continue to the next page instead?

**Comment by marc-thieme** [comment]
Hi, thank you very much for using the package and the suggestion.

I agree with you that this functionality would be useful. I spend a lot of time trying to figure out how to do it. I prepared a proposal on branch [1_add-breakable-show-rule](https://github.com/marc-thieme/frame-it/tree/1_add-breakable-show-rule). Could you please take a look at the code readme on that branch. I demonstrate the new syntax at the very end of the document. 

Could you please give me feedback regarding:
- If this is solving your problem/what you asked for in a satisfactory way (very important point for me!)
- The syntax
- The layout / result
- How understandable the approach is based on the explanation, naming and the readme

I would appreciate it a lot! Thank you!

## Some of my considerations
I would strongly like to avoid adding new parameters to the `init-frames` function because I want to be as beginner friendly and non-overwhelming as possible.

Additionally imo, the ergonomics as a show rule fit the theme application well.

Most importantly and rather technically though, we need an api using a show rule for this anyway. Let me explain.
Any concrete function (i.e. a `definition(...)`–function) must return a type `figure`. Otherwise (i.e. if it is type `content`), typst does not allow sticking a label `<...>` onto it afterwards, which is a hard requirement for me. 
However, when you try to make that figure breakable from inside the function, then you need to wrap the figure with a `show` rule. However, this turns the returned type into a generic `content`, which makes it unlabellable.

**Comment by just-bqn** [comment]
Hi!

The syntax and explanation are good. For my use case, I simply add the show rule at the start of my notes. The solution works well, even when I insert figures like tables and pictures. If anything,

- It can be difficult to tell whether a frame at the start of a page is a continuation from the previous page or a new headless frame (you can tell by their content or the rounded corners, but the difference can be quite subtle). Maybe the top and bottom bars can be made visible only for the first and last frames, like [this](https://tex.stackexchange.com/a/494276). Though I'm not sure if that would be the most aesthetically pleasing solution...
- It seems like `styles.hint` is not supported yet.

<details>
  <summary>Image</summary>
  
![image](https://github.com/user-attachments/assets/2b8de341-7b67-4b4c-9d6c-cb6721d2cc04)

</details>

<details>
  <summary>Code</summary>
  
  ```
  #import "frame-it/src/lib.typ": *
  
  #let (theorem,) = make-frames(
    "core-frames",
    theorem: ("Theorem",),
  )
  
  #show: breakable-frames("core-frames")
  #set page(height: 200pt)
  
  #theorem[A very long theorem][
    #lorem(200)
  ]
  
  #theorem[A theorem with a table and a picture][
    #table(
      columns: 2,
      [#lorem(2)], [#lorem(5)],
      [#lorem(2)], [#lorem(5)],
      [#lorem(2)], [#lorem(5)],
      [#lorem(2)], [#lorem(5)],
      [#lorem(2)], [#lorem(5)],
    )
  
    #image("image.png", height: 80pt) // https://typst.app/docs/reference/visualize/image/
  ]
  
  #theorem(style: styles.hint)[Another very long theorem][
    #lorem(100)
  ]
  ```

</details>

Thanks for the quick response! The frames are beautiful. ദ്ദി ( ᵔ ᗜ ᵔ )

**Comment by marc-thieme** [comment]
![image](https://github.com/user-attachments/assets/691dffdf-c240-4e68-89c0-3142f61fcdab)
How would you like this compromise?
Here, I simply place the same line again on top of the other but this time starting from the bottom going up.
This also means that it does not work if the hint is spanning more than 2 different pages because we only draw the line on the first and last page.

However, I would pose to ignore this issue :sweat_smile: as I don't expect such large hints to be a use case. Feedback by users appreciated here on how you are using the library and if you have such large hints in your documents.

If you need that I would next try to layer a colored border on the box yet again over the two lines. However, I imagine this to be a little fiddly and time–consuming. (Btw. I'm using explicitly placed lines because you would not be able to round the ends when using the border of the block containing the text to draw the lines.

Lastly, since this only concerns the implementation of the hint style I feel like somebody else could fiddle around to achieve this if need be. I would be glad to support this effort.

I pushed the added line to the branch but I'm open to end up with a different approach in the end.

**Comment by just-bqn** [comment]
I agree that such a long hint would be quite niche. Personally I'll settle with not using explicitly placed lines and give up on the rounded corners lol.
  
![image](https://github.com/user-attachments/assets/a56c9708-f768-489a-b1c1-8893d7287acb)

<details>
  <summary>Code</summary>
  
  ```
  // src/styles.typ
  
  #let hint_alt(title, tags, body, supplement, number, accent-color) = {
    let stroke = accent-color + 3.5pt
  
    let header = if title == none {
      none
    } else {
      if title != [] {
        title = [~~#title~]
      }
  
      let tag-str = if tags != () {
        [~(#tags.join(", "))~]
      } else {
        []
      }
      let supplement-str = text(gray.darken(50%))[#supplement #number]
      let head-body-separator = if body == [] [] else [:]
      [~#supplement-str~*#(title)*_#(tag-str)_#head-body-separator~]
    }
  
    align(
      left,
      block(
        width: 100%,
        inset: body-inset,
        stroke: (
          left: stroke,
        ),
        spacing: 0em,
        outset: (y: 0em),
        header + body,
      ),
    )
  }
  ```
  
  ```
  #import "frame-it/src/lib.typ": *
  
  #let (theorem,) = make-frames(
    "frames",
    theorem: ("Theorem",),
  )
  
  #show: breakable-frames("frames")
  #set page(height: 200pt)
  
  #theorem(style: styles.hint_alt)[A theorem][
    #lorem(50)
  ]
  
  #theorem(style: styles.hint_alt)[A very long theorem][
    #lorem(300)
  ]
  ```

</details>

**Comment by marc-thieme** [comment]
Hi thanks for your comment. This just gave me the impulse to give it a go still to account for all possibilities. Turns out that it was in fact very easy.

The new layouting just places two very small lines at the tips of the border. This smoothes only the tips which are concluding the hint. Here is a schematic:
![image](https://github.com/user-attachments/assets/043174c2-6393-43ed-96be-a0c9fda1baf3)

If you prefer the hint without the rounding, by all means stick to implementing the style yourself. I always wondered if anybody would ever use the custom styling feature haha.
If not, I am pretty confident in this version. If you agree, I will merge it into main later. Until I release a new version (I think I'm gonna wait) you will have to use your local version of the library anyway.

**Comment by marc-thieme** [comment]
Closed by #2
